ale/base/type_sensor.py

```diff
@@ -362,3 +368,125 @@ def line_times(self):
+            self._cahvor_rotation_matrix = np.zeros((3, 3))
+            self._cahvor_rotation_matrix[0, 0] = math.cos(phi) * math.cos(k)
+            self._cahvor_rotation_matrix[0, 1] = math.sin(w) * math.sin(phi) * math.cos(k) + \
+                math.cos(w) * math.sin(k)
+            self._cahvor_rotation_matrix[0, 2] = - math.cos(w) * math.sin(phi) * math.cos(k) + \
+                math.sin(w) * math.sin(k)
+            self._cahvor_rotation_matrix[1, 0] = - math.cos(phi) * math.sin(k)
+            self._cahvor_rotation_matrix[1, 1] = - math.sin(w) * math.sin(phi) * math.sin(k) + \
+                math.cos(w) * math.cos(k)
+            self._cahvor_rotation_matrix[1, 2] = math.cos(w) * math.sin(phi) * math.sin(k) + \
+                math.sin(w) * math.cos(k)
+            self._cahvor_rotation_matrix[2, 0] = math.sin(phi)
+            self._cahvor_rotation_matrix[2, 1] = - math.sin(w) * math.cos(phi)
+            self._cahvor_rotation_matrix[2, 2] = math.cos(w) * math.cos(phi)
+        return self._cahvor_rotation_matrix
+
+    @property
+    def frame_chain(self):
+        """
+        Returns a modified frame chain with the cahvor models extra rotation
+        added into the model
+
+        Returns
+        -------
+        : object
+          A networkx frame chain object
+        """
+        if not hasattr(self, '_frame_chain'):
+            self._frame_chain = FrameChain.from_spice(sensor_frame=self.ikid,
+                                                      target_frame=self.target_frame_id,
+                                                      center_ephemeris_time=self.center_ephemeris_time,
+                                                      ephemeris_times=self.ephemeris_time,
+                                                      nadir=False, exact_ck_times=False)
+            cahvor_quats = Rotation.from_matrix(self.cahvor_rotation_matrix).as_quat()
+            cahvor_rotation = ConstantRotation(cahvor_quats, self.sensor_frame_id, self.ikid)
+            self._frame_chain.add_edge(rotation = cahvor_rotation)
+        return self._frame_chain
+
+    @property
+    def detector_center_line(self):
+        """
+        Computes the detector center line using the cahvor model.
+        Equation for computation comes from MSL instrument kernels
+
+        Returns
+        -------
+        : float
+          The detector center line/boresight center line
+        """
+        return np.dot(self.cahvor_camera_dict['V'], self.cahvor_camera_dict['A'])
+
+    @property
+    def detector_center_sample(self):
+        """
+        Computes the detector center sample using the cahvor model.
+        Equation for computation comes from MSL instrument kernels
+
+        Returns
+        -------
+        : float
+          The detector center sample/boresight center sample
+        """
+        return np.dot(self.cahvor_camera_dict['H'], self.cahvor_camera_dict['A'])
+
+    @property
+    def pixel_size(self):
```

Comment:
We should add properties for focal2pixel_lines and focal2pixel_samples that use this. focal2pixel_lines is (0, 1/pixel_size, 0) and focal2pixel_samples is (1/pixel_size, 0, 0)

Reply:
This currently resides on the concrete driver, should it be part of the more generic mixin?